$NetBSD: patch-ap,v 1.4 2013/10/27 21:21:26 dholland Exp $

https://sourceforge.net/tracker/?func=detail&aid=3018741&group_id=21302&atid=121302
(there's a leftover bit apparently missed upstream)

Also, add <cstring> to fix build with png 1.6.

--- src/celengine/image.cpp.orig	2011-06-05 16:11:09.000000000 +0000
+++ src/celengine/image.cpp
@@ -7,6 +7,7 @@
 // as published by the Free Software Foundation; either version 2
 // of the License, or (at your option) any later version.
 
+#include <cstring>
 #include <fstream>
 
 #ifndef TARGET_OS_MAC
@@ -78,7 +79,7 @@ using namespace std;
 // Define various expansion transformations for old versions of libpng
 #if PNG_LIBPNG_VER < 10004
 #define png_set_palette_to_rgb(p)  png_set_expand(p)
-#define png_set_gray_1_2_4_to_8(p) png_set_expand(p)
+#define png_set_expand_gray_1_2_4_to_8(p) png_set_expand(p)
 #define png_set_tRNS_to_alpha(p)   png_set_expand(p)
 #endif // PNG_LIBPNG_VER < 10004
 
